src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -198,7 +198,36 @@ protected Solution SetCurrentSolution(Solution solution)
         /// <param name="projectId">The id of the project updated by <paramref name="transformation"/> to be passed to the workspace change event.</param>
         /// <param name="documentId">The id of the document updated by <paramref name="transformation"/> to be passed to the workspace change event.</param>
         /// <returns>True if <see cref="CurrentSolution"/> was set to the transformed solution, false if the transformation did not change the solution.</returns>
-        internal bool SetCurrentSolution(Func<Solution, Solution> transformation, WorkspaceChangeKind kind, ProjectId? projectId = null, DocumentId? documentId = null)
+        internal bool SetCurrentSolution(
+            Func<Solution, Solution> transformation,
+            WorkspaceChangeKind kind,
+            ProjectId? projectId = null,
+            DocumentId? documentId = null)
+        {
+            return SetCurrentSolution(
+                transformation,
+                onAfterUpdate: (oldSolution, newSolution) =>
```

Comment:
This feels like a hot path, consider using `PooledDelegates` maybe?

Reply:
i can actually just have it pass the data needed to the action.  i'll make that change

Reply:
done.